internal/impl/kafka/enterprise/redpanda_migrator_offsets_input.go

```diff
@@ -147,16 +146,18 @@ func init() {
 					return nil, fmt.Errorf("failed to decode record key: %s", err)
 				}
 
-				var isExpectedTopic bool
-				if len(topicPatterns) > 0 {
-					isExpectedTopic = slices.ContainsFunc(topicPatterns, func(tp *regexp.Regexp) bool {
-						return tp.MatchString(key.Topic)
-					})
-				} else {
-					isExpectedTopic = slices.ContainsFunc(topics, func(t string) bool {
-						return t == key.Topic
+				matchesTopic := func(topic string) bool {
+					if len(topicPatterns) > 0 {
+						return slices.ContainsFunc(topicPatterns, func(tp *regexp.Regexp) bool {
+							return tp.MatchString(topic)
+						})
+					}
+					return slices.ContainsFunc(topics, func(t string) bool {
+						return t == topic
```

Comment:
good! Now I think this can be lifted out of this closure so it's declared before `NewFranzReaderOrderedFromConfig` is called. It helps remove clutter from what this function is doing.

Reply:
Good point! Sorted ❤️